src/components/TextInput/TextInput.test.tsx

```diff
@@ -0,0 +1,292 @@
+
+    expect(changeHandler.callCount).to.equal(1);
+  });
+});
+
+describe("Renders TextInput with auto-generated ID, hidden label and visible helper text", () => {
+  let container;
+  let generateUUIDSpy;
+
+  before(() => {
+    generateUUIDSpy = spy(generateUUID);
+    container = Enzyme.mount(
+      <>
+        <TextInput
+          labelText={"Custom Input Label"}
+          showLabel={false}
+          helperText={"Custom Helper Text"}
+          required={true}
+          placeholder={"Input Placeholder"}
+          type={TextInputTypes.text}
+        ></TextInput>
+      </>
+    );
+  });
+
+  it("Renders Input component", () => {
+    expect(container.find("input").exists()).to.equal(true);
+  });
+
+  it("Calls a UUID generation method if no ID is passed as a prop", () => {
+    expect(container.find("input").props()).to.have.property("id");
+    expect(generateUUIDSpy.called);
+  });
+
+  it("Does not renders Label component", () => {
+    expect(container.find("label").exists()).to.equal(false);
+  });
+
+  it("Renders custom aria-label", () => {
+    expect(container.find("input").prop("aria-label")).to.equal(
+      "Custom Input Label"
+    );
+  });
+
+  it("Renders HelperErrorText component", () => {
+    expect(container.find(".helper-text").exists()).to.equal(true);
+  });
+
+  it("Renders aria-describedby with helper text", () => {
+    expect(container.find("input").prop("aria-describedby")).to.equal(
+      "Custom Helper Text"
+    );
+  });
+});
+
+describe("TextInput shows error state", () => {
+  let container;
+  before(() => {
+    container = Enzyme.mount(
+      <>
+        <TextInput
+          id={"myTextInputError"}
+          labelText={"Custom Input Label"}
+          helperText={"Custom Helper Text"}
+          errorText={"Custom Error Text"}
+          placeholder={"Input Placeholder"}
+          errored={true}
+          type={TextInputTypes.text}
+        ></TextInput>
+      </>
+    );
+  });
+
+  it("Renders Input component", () => {
+    expect(container.find("input").exists()).to.equal(true);
+  });
+
+  it("Renders Label component", () => {
+    expect(container.find("label").exists()).to.equal(true);
+  });
+
+  it("Renders HelperErrorText component", () => {
+    expect(container.find(".helper-text").exists()).to.equal(true);
+  });
+
+  it("Input shows error state", () => {
+    expect(container.find("input").hasClass("inputfield--error")).to.equal(
+      true
+    );
+  });
+
+  it("Helper text shows error state", () => {
+    expect(
+      container.find(".helper-text").hasClass("helper-text--error")
+    ).to.equal(true);
+  });
+
+  it("Helper text shows error text", () => {
+    expect(container.find(".helper-text").text()).to.equal("Custom Error Text");
+  });
+});
+
+describe("TextInput Group", () => {
+  let container;
+  before(() => {
+    container = Enzyme.mount(
+      <fieldset>
+        <legend>Input Group Label</legend>
+        <>
+          <>
```

Comment:
Only the outer fragment is needed.  The child components don't need it. For example:
```<><TextInput></TextInput><TextInput></TextInput></>```

Reply:
@kaplanmichael 
Agreed.  Fixed.

Reply:
Looking more closely, I see that you actually don't need the fragment tags at all here since you already have a valid wrapping element (fieldset in this case).

Reply:
Again, good suggestion.  This has been fixed.